src/shared/components/plots/ScatterPlot.tsx

```diff
@@ -329,6 +340,44 @@ export default class ScatterPlot<D extends IBaseScatterPlotData> extends React.C
         );
     }
 
+    @computed get regressionLineComputations() {
+        return regression.linear(this.props.data.map(
+            // perform same transformations on data for this calculation as we do for plot (i.e. log if necessary)
+            d=>([this.x(d), this.y(d)]))
+        );
+    }
+
+    private get regressionLine() {
+        if (this.props.showRegressionLine) {
+            const coeffs = this.regressionLineComputations.equation;
```

Comment:
it seems like there's some actual computation here that ought to be unit tested.